sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/eventlogger/EventsSender.kt

```diff
@@ -0,0 +1,57 @@
+package com.rakuten.tech.mobile.sdkutils.eventlogger
+
+import okhttp3.ResponseBody
+import retrofit2.Call
+import retrofit2.http.Body
+import retrofit2.http.Header
+import retrofit2.http.POST
+import java.io.IOException
+
+internal interface EventsSender {
+
+    /**
+     * Thread-blocking operation that sends the supplied [events] to backend and invokes the optional [onSuccess]
+     * callback if succeeded.
+     */
+    fun pushEvents(events: List<Event>, onSuccess: (() -> Unit)? = null)
```

Comment:
what about if we add @WorkerThread annotation in this function?

Reply:
Thanks @mdusmangani-jp, I didn't know this annotation exists. I think it is good for readability. Let me add